WooCommerce/Classes/ViewModels/Order Details/OrderDetailsViewModel.swift

```diff
@@ -557,32 +573,30 @@ extension OrderDetailsViewModel {
         stores.dispatch(action)
     }
 
-    func syncShippingLabels(onCompletion: ((Error?) -> ())? = nil) {
-        // If the plugin is not active, there is no point on continuing with a request that will fail.
-        isPluginActive(SitePlugin.SupportedPlugin.WCShip) { [weak self] isActive in
-
-            guard let self = self, isActive else {
-                onCompletion?(nil)
-                return
-            }
-
-            let action = ShippingLabelAction.synchronizeShippingLabels(siteID: self.order.siteID, orderID: self.order.orderID) { result in
+    @MainActor
+    func syncShippingLabels() async {
+        guard orderContainsOnlyVirtualProducts == false else {
+            return
+        }
+        guard await isPluginActive(SitePlugin.SupportedPlugin.WCShip) else {
+            return
+        }
```

Comment:
nit: These could be combined in a single `guard` statement, right?

Reply:
Good catch, missed this one! Updated in https://github.com/woocommerce/woocommerce-ios/pull/10273/commits/fdfb32c9162592b4326b54034f0657ad65f07f6d